### PR TITLE
Set up mongoDB database for filesystem

### LINKED ISSUE #3: FileSystem
Ability to store file data as Javascript Object

### PR DESCRIPTION
**Pull request by ankushbhardwxj**
Resolves #3 
This is a python script that can create a custom `fs.json` file like we had created before. 
This would be a great solution for the filesystem. 
How this works?
1. We present user with a `Welcome Screen` like every other OS, where we ask users to setup the FS.
2. Users create a new directory locally, and can put up any number of directories and files inside it.
3. Then users run this python script taking input as that directory, and then it gives output as a `.json` file.
4. This `fs.json` file serves as our filesystem and we can access it using terminal or the file explorer.

@RAJAGOPALAN-GANGADHARAN  Please review and merge  👍

### CODE REVIEW

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
Awesome but How is this invocated? also why cant we do this in JS itself(It would be better to keep it mono 😄 )

**Comment by ankushbhardwxj** [comment]
I tried to get it done in javascript but failed. Python just makes it very easy. 
This can be invoked currently by running the python command locally. But we can try to get rid of that.

**Comment by ankushbhardwxj** [comment]
@RAJAGOPALAN-GANGADHARAN 
I hope you understood how it works.
- We create a local filesystem.
- Then we run the python script to convert it to the `.json` format.

**Comment by ankushbhardwxj** [comment]
@RAJAGOPALAN-GANGADHARAN If you insist on keeping it mono, please take a look at https://www.npmjs.com/package/directory-tree . This package does the same thing. But I'm not sure if we can use it.

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
This looks great see if you can integrate it here. Will be great if we keep it uniform.

**Comment by ankushbhardwxj** [comment]
@RAJAGOPALAN-GANGADHARAN I integrated it. It works fine. But it doesn't store contents of a file. It stores the filename, type and other attributes but no contents. The python script also stores the contents , however this doesn't. Another problem is that it shows the json file in console only. It doesnt create a separate file with `json` format. I think we should move forward with the python script.

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
So it stores filenames right. Use that to our advantage - Use the filename as a query id( maybe use mongoDB) to query the contents ( like extension,contents) That will give a lot of flexibility. For eg Makes easy API control . Try keeping it within the ecosystem. Thanks!

**Comment by ankushbhardwxj** [comment]
Oh. So you wanted to use mongoDB and NodeJS? If thats the case we dont need all these at all. I thought you wanted to have a single .json file which would constitute the filesystem.

**Comment by ankushbhardwxj** [comment]
This wont be efficient according to me. Imagine making api calls every time we type a command on the terminal.

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
We can later find a way to cache stuff. But having a well organized fs is important. Putting everything in json file is a bad practice. So try using mongoDB as our fs, it will be pretty manageable. Also there are some caching techniques which we will see later to improve performance. This is also to attempt to offload some work off the browser. I am not sure if json are stored in browser cache but its probably good practice to load work off the browser as its already doing the heavy lifting. Kudos for ur ideas 👍

**Comment by ankushbhardwxj** [comment]
@RAJAGOPALAN-GANGADHARAN I have set up an express server and have defined the filesystem schema. Please review and merge 👍

File: src/server/database/file.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        type : String
+    },
+    directory : {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "Directory"
+    }
+})
+
+const File = mongoose.model("File", fileSchema)
+module.exports = File
```
Comment (RAJAGOPALAN-GANGADHARAN) [inline src/server/database/file.js]: can u add a new line at the end of every file please. Git requires it to determine the line ending type.

Comment (ankushbhardwxj) [reply src/server/database/file.js]: done

Comment (RAJAGOPALAN-GANGADHARAN) [reply src/server/database/file.js]: Awesome added 20 points

Comment (ankushbhardwxj) [reply src/server/database/file.js]: Thank you so much

**Comment by ankushbhardwxj** [comment]
@RAJAGOPALAN-GANGADHARAN why isn't this merged ? Are there any changes to be made ?

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
Done please continue 😃

**Pull request merged**